pkg/registry/resource/resourceclaim/declarative_validation_test.go

```diff
@@ -330,10 +382,73 @@ func mkResourceClaimWithStatus(tweaks ...func(rc *resource.ResourceClaim)) resou
 	return rc
 }
 
+func mkResourceClaimWithDevicesStatus(tweaks ...func(rc *resource.ResourceClaim)) resource.ResourceClaim {
```

Comment:
There's a lot of duplication in these - should they just be chained?  Or should basic status be poart of the "main" constructor?  Or represented as tweaks in the first place?

e.g. 

```
mkValidResourceClaim(
    tweakSetStatus,
    tweakSetDeviceStatus,
)
```